src/webgl/p5.RendererGL.js

```diff
@@ -32,85 +34,114 @@ defineStrokeJoinEnum('ROUND', 0);
 defineStrokeJoinEnum('MITER', 1);
 defineStrokeJoinEnum('BEVEL', 2);
 
-const lightingShader = readFileSync(
+const getCachedShader= (shaderKey, shaderSource)=>{
```

Comment:
I think this would need to be not just a `const`, but a property on `p5.RendererGL`, as each instance of a renderer will need its own shaders. We just want the ability to share those shaders within the renderer.